src/services/withDefault.ts

```diff
@@ -0,0 +1,16 @@
+import { asParamGetSet } from '@/services/asParamGetSet'
+import { ExtractParamType, isParamGetSet } from '@/types/params'
+import { Param, ParamGetSet } from '@/types/paramTypes'
+
+export type ParamWithDefault<TParam extends Param, TParamType extends ExtractParamType<TParam> = ExtractParamType<TParam>> = Required<ParamGetSet<TParamType>>
```

Comment:
Seems like we don't need the second generic? That can just be part of the return type right?

Reply:
good call, found a couple more than just this file as well 👍